examples/armv4t/gdb/host_io.rs

```diff
@@ -0,0 +1,236 @@
+use gdbstub::target;
+use std::io::{Read, Seek, Write};
+
+use crate::emu::{Emu, FD_MAX};
+
+use gdbstub::target::ext::host_io::{
+    FsKind, HostIoErrno, HostIoError, HostIoOpenFlags, HostIoOpenMode, HostIoOutput, HostIoResult,
+    HostIoStat, HostIoToken,
+};
+
+impl target::ext::host_io::HostIo for Emu {
+    #[inline(always)]
+    fn enable_open(&mut self) -> Option<target::ext::host_io::HostIoOpenOps<Self>> {
+        Some(self)
+    }
+
+    #[inline(always)]
+    fn enable_close(&mut self) -> Option<target::ext::host_io::HostIoCloseOps<Self>> {
+        Some(self)
+    }
+
+    #[inline(always)]
+    fn enable_pread(&mut self) -> Option<target::ext::host_io::HostIoPreadOps<Self>> {
+        Some(self)
+    }
+
+    #[inline(always)]
+    fn enable_pwrite(&mut self) -> Option<target::ext::host_io::HostIoPwriteOps<Self>> {
+        Some(self)
+    }
+
+    #[inline(always)]
+    fn enable_unlink(&mut self) -> Option<target::ext::host_io::HostIoUnlinkOps<Self>> {
+        Some(self)
+    }
+
+    #[inline(always)]
+    fn enable_readlink(&mut self) -> Option<target::ext::host_io::HostIoReadlinkOps<Self>> {
+        Some(self)
+    }
+
+    #[inline(always)]
+    fn enable_setfs(&mut self) -> Option<target::ext::host_io::HostIoSetfsOps<Self>> {
+        Some(self)
+    }
+}
+
+impl target::ext::host_io::HostIoOpen for Emu {
+    fn open(
+        &mut self,
+        filename: &[u8],
+        flags: HostIoOpenFlags,
+        _mode: HostIoOpenMode,
+    ) -> HostIoResult<u32, Self> {
+        let path = match std::str::from_utf8(filename) {
+            Ok(v) => v,
+            Err(_) => return Err(HostIoError::Errno(HostIoErrno::ENOENT)),
+        };
```

Comment:
Doesn't rust have a universal way to handle with filename?

Reply:
Unlike other languages, which try to pave over platform differences when possible (Go is one example that comes to mind), Rust takes the approach of "abstract when possible, but use platform-specific behavior when a reasonable abstraction is impossible".

Essentially, Rust gives you the flexibility to be as "correct" as you'd like, without implicitly locking you in to some kind of built-in abstraction.

The "easy" approach that many applications use is to convert paths into `str`, and then wrap those in a `std::path::Path`. I'd wager that this works fine 95% of the time, which is why I don't mind taking this approach here.

It's moreso that there _are_ some file paths that aren't valid `str`s on Unix/Windows, and to be 100% correct, there should be platform-specific logic to convert raw `&[u8]` buffers into platform-specific `std::ffi::OsStr` values (which can then be wrapped in `std::path::Path`).

...but that's hard, and the extra 5% might not be worth it. I just thought I'd point it out, moreso as an opportunity for "learning" rather than as something we'd actually want to dig into / properly handle.